src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -1,20 +1,39 @@
 package lotto.domain;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LottoNumber {
-    private final int number;
+    private static final int MIN = 1;
+    private static final int MAX = 45;
+    private static final Map<Integer, LottoNumber> CACHE_NUM = new HashMap<>();
 
-    public LottoNumber(String number) {
-        this(Integer.parseInt(number));
+    static{
+        for (int i = MIN; i <= MAX; i++){
+            CACHE_NUM.put(i, new LottoNumber(i));
+        }
     }
 
+    private final int number;
+
     public LottoNumber(int number) {
         validation(number);
         this.number = number;
     }
-    private void validation(int number) {
-        if (number < 1 || number > 45) {
+
+    public static LottoNumber valueOf(String number){
+        validation(Integer.parseInt(number));
+        return CACHE_NUM.get(Integer.parseInt(number));
```

Comment:
```suggestion
        return valueOf(Integer.parseInt(number));
```
메서드 오버로딩시 생성자와 같이 다른 메서드를 호출해 구현하면 중복 제거효과